lib/infrastructure/display_qr_code/qr_seed_repository.dart

```diff
@@ -0,0 +1,60 @@
+import 'package:injectable/injectable.dart';
+import 'package:superformula_mobile_test/domain/core/exceptions.dart';
+import 'package:superformula_mobile_test/domain/display_qr_code/entities/qr_seed.dart';
+import 'package:superformula_mobile_test/domain/core/either.dart';
+import 'package:superformula_mobile_test/domain/display_qr_code/i_qr_seed_local_data_source.dart';
+import 'package:superformula_mobile_test/domain/display_qr_code/i_qr_seed_remote_data_source.dart';
+import 'package:superformula_mobile_test/domain/display_qr_code/i_qr_seed_repository.dart';
+import 'package:superformula_mobile_test/domain/display_qr_code/qr_seed_failure.dart';
+import 'package:superformula_mobile_test/domain/platform/i_network_info.dart';
+
+import '../../locator.dart';
+
+@Injectable(as: IQrSeedRepository)
+class QrSeedRepository implements IQrSeedRepository {
+  @override
+  Future<Either<QrSeedFailure, QrSeed>> getQrCodeSeed() async {
+    final remoteDataSource = locator<IQrCodeRemoteDataSource>();
+    final localDataSource = locator<IQrCodeLocalDataSource>();
+    final networkInfo = locator<INetworkInfo>();
```

Comment:
That actually was left this way intentionally. For the sole purposes of this code challenge, my intentions were to demonstrate that both ways could be done. I prefer to move the dependencies to the class level as well (since we can even scope it better).